internal/pkg/db/postgres/utils.go

```diff
@@ -128,7 +130,19 @@ func sortedSqlFileNames(sqlFilesDir string) ([]string, errors.EdgeX) {
 func WrapDBError(message string, err error) errors.EdgeX {
 	var pgErr *pgconn.PgError
 	if goErrors.As(err, &pgErr) {
+		if pgerrcode.IsIntegrityConstraintViolation(pgErr.Code) {
+			return errors.NewCommonEdgeX(errors.KindDuplicateName, pgErr.Detail, nil)
+		}
 		return errors.NewCommonEdgeX(errors.KindDatabaseError, fmt.Sprintf("%s: %s %s", message, pgErr.Error(), pgErr.Detail), nil)
 	}
 	return errors.NewCommonEdgeX(errors.KindDatabaseError, message, err)
 }
+
+// ConvertMapToJSONString parses the query map to a JSON string to be used in the sql statement
+func ConvertMapToJSONString(fieldMap map[string]any) (string, errors.EdgeX) {
```

Comment:
move this function to internal/pkg/utils/struct.go , because this is not postgres specific

Reply:
This function is removed since the table design has been changed.